internal/broker/provision.go

```diff
@@ -294,12 +295,21 @@ func getNamespaceFromContext(contextProfile map[string]interface{}) (internal.Na
 }
 
 func createReleaseName(name internal.AddonName, planName internal.AddonPlanName, iID internal.InstanceID) internal.ReleaseName {
-	maxLen := 53
-	relName := fmt.Sprintf("hb-%s-%s-%s", name, planName, iID)
-	if len(relName) <= maxLen {
-		return internal.ReleaseName(relName)
+	// max name length 53 = 36(GUID) + 4(pre + special char) + 12 (6 for name, 6 for plan) + 1 extra char
+	maxLen := 6
+	if len(name) > maxLen {
```

Comment:
nit: you can have just one function , consider this
```go
func normalize(name string) string {
	maxLen := 6
	if len(name) > maxLen {
		name = name[:maxLen]
	}
	return strings.Trim(name, "-")
}

func createReleaseName(name internal.AddonName, planName internal.AddonPlanName, iID internal.InstanceID) internal.ReleaseName {
	// max name length 53 = 36(GUID) + 4(pre + special char) + 12 (6 for name, 6 for plan) + 1 extra char
	releaseName := fmt.Sprintf(
		"hb-%s-%s-%s",
		normalize(string(name)),
		normalize(string(planName)),
		iID)

	return internal.ReleaseName(releaseName)
}
```
thanks to that you will have just one function where are you doing some normalization. In current approach it is split.